packages/amazon/src/serverGroup/details/scalingPolicy/upsert/ScalingPolicyCommandBuilderService.ts

```diff
@@ -0,0 +1,127 @@
+import { cloneDeep } from 'lodash';
+
+import {
+  ISimplePolicyDescription,
+  IStepPolicyDescription,
+  IUpsertAlarmDescription,
+  IUpsertScalingPolicyCommand,
+} from '../ScalingPolicyWriter';
+import { IAmazonServerGroup, IScalingPolicy, IStepAdjustment, ITargetTrackingPolicy } from '../../../../domain';
+
+type PolicyType = 'Step' | 'TargetTracking';
+
+export const ScalingPolicyCommandBuilder = {
+  buildAlarm: (policy: IScalingPolicy, region: string, asgName: string): IUpsertAlarmDescription => {
+    const alarm = policy?.alarms[0];
```

Comment:
This will blow up if `alarm` is undefined; should we account for it?

Reply:
The policy command builder (new) or api response (for existing) always sets it to an empty array, but it never hurts to do an additional safety check so I will add that.